scripts/cache-olean.py

```diff
@@ -87,6 +87,12 @@ def fetch_mathlib(asset):
         print('Repository not initialized')
         sys.exit(-1)
 
+    if repo.is_dirty():
```

Comment:
I think the flag should only work with `--fetch`. And with fetch when it is absent and the directory is dirty, I would print an error message and exit with code -1

With `build`, `build-all` and without flag (i.e. `cache`) if the directory is dirty, I would print an error and suggest the user commits the changes before building without possibility of bypassing the behavior. Because the cache is indexed with the hash code of the commit, if the directory is dirty, we're storing the wrong files.

Reply:
Thanks for the feedback, I agree this would be better. I have just added a commit which attempts to implement the scheme you have sketched. :-)